test/main.cc

```diff
@@ -0,0 +1,151 @@
+#include <random>
+
+#include "../src/class.h"
+#include "gtest/gtest.h"
+
+const int kDefaultWidth = 80;
+const int kDefaultHeight = 60;
+
+class WebPSamplesGenerator {
+ public:
+  // Initializing.
+  WebPSamplesGenerator()
+      : pic_count(10),
+        width(kDefaultWidth),
+        height(kDefaultHeight),
+        transparency(0xff),
+        randomized(true) {}
+
+  WebPSamplesGenerator(int pic_count, uint8_t transparency, bool randomized)
+      : pic_count(pic_count),
+        width(kDefaultWidth),
+        height(kDefaultHeight),
+        transparency(transparency),
+        randomized(randomized) {}
+
+  WebPSamplesGenerator(int pic_count, int width, int height,
+                       uint8_t transparency, bool randomized)
+      : pic_count(pic_count),
+        width(width),
+        height(height),
+        transparency(transparency),
+        randomized(randomized) {}
+
+  // Returns RGBA values for WebPPicture.
+  std::vector<uint8_t> GenerateRGBA(int seed) {
+    std::mt19937 rng(seed);
+    std::vector<uint8_t> rgba;
+
+    int color_R = rng();
```

Comment:
This is because I want the color to be different on each frame (so that I know it's an animation).

Reply:
Yes it doesn't have to be a file constant, but it can still be a const variable within the function, if it's only assigned once.. The same goes for other variables like pic_count, transparency, etc. If a variable is only assigned once, it can be marked const.